main/src/components/docs/LegendItemWithSplit.jsx

```diff
@@ -0,0 +1,201 @@
+    groupElements.forEach(groupElement => {
+      const newSvg = document.createElementNS('http://www.w3.org/2000/svg', 'svg');
+      const attributesToExclude = [WIDTH_ATTR, HEIGHT_ATTR, VIEW_BOX_ATTR];
+      for (const attr of originalSvgElement.attributes) {
+        if (!attributesToExclude.includes(attr.name)) {
+          newSvg.setAttribute(attr.name, attr.value);
+        }
+      }
+      if (originalDefs) {
+        newSvg.appendChild(originalDefs.cloneNode(true));
+      }
+      originalStyles.forEach(styleNode => {
+        newSvg.appendChild(styleNode.cloneNode(true));
+      });
+      const clonedGroupElement = groupElement.cloneNode(true);
+      if (clonedGroupElement.hasAttribute(TRANSFORM_ATTR)) {
+        clonedGroupElement.removeAttribute(TRANSFORM_ATTR);
+      }
+      newSvg.appendChild(clonedGroupElement);
+      const serializer = new XMLSerializer();
+      let svgString = serializer.serializeToString(newSvg);
+      const groupBBox = measurementComponentRef.current.measureGroupBBox(svgString);
+      const width = groupBBox.width.toFixed(2);
+      const height = groupBBox.height.toFixed(2);
+      newSvg.setAttribute(WIDTH_ATTR, width);
+      newSvg.setAttribute(HEIGHT_ATTR, height);
+      newSvg.setAttribute(VIEW_BOX_ATTR, `${groupBBox.x.toFixed(2)} ${groupBBox.y.toFixed(2)} ${width} ${height}`)
+      svgString = serializer.serializeToString(newSvg);
+      const title = svgPartsMap.get(groupElement.id)
+      svgArray.push({ id: groupElement.id || `group-${svgArray.length}`, svgString, title });
+    });
+    return svgArray;
+  }
+
+  const useSplitSvg = (svgContent, setSplitSvgs) => {
+    useEffect(() => {
+      if (svgContent) {
+        if (svgCache.has(svgContent.svgPath)) {
+          setSplitSvgs(svgCache.get(svgContent.svgPath));
+          return;
+        }
+        try {
+          const svgArray = splitSvgToArray(svgContent.data);
+          svgCache.set(svgContent.svgPath, svgArray);
+          setSplitSvgs(svgArray);
+        } catch (e) {
+          setSplitSvgs([]);
+        }
+      } else {
+        setSplitSvgs([]);
+      }
+    }, [svgContent]);
+  };
+
+  useSplitSvg(svgContentDay, setSplitSvgsDay);
+  useSplitSvg(svgContentNight, setSplitSvgsNight);
+
+  const chunkArray = (arr, chunkSize) => {
+    const R = [];
+    for (let i = 0; i < arr.length; i += chunkSize) {
+      R.push(arr.slice(i, i + chunkSize));
+    }
+    return R;
+  };
+
+  const rowsDay = chunkArray(splitSvgsDay, COLUMN_COUNT);
+  const rowsNight = chunkArray(splitSvgsNight, COLUMN_COUNT);
+
+  const SvgTable = ({ rows, mode }) => {
+    const isNightMode = mode === 'night';
+    const hasRows = isNightMode ? rowsNight.length > 0 : rowsDay.length > 0;
+    const isLoading = isNightMode ? loadingNight : loadingDay;
+    const hasContent = isNightMode ? !svgContentNight && !loadingNight : !svgContentDay && !loadingDay
+    const legendStyle = isNightMode ? styles.legendNight : styles.legendDay;
+
+    if (hasRows) {
```

Comment:
For images you can use lazy loading. For your SVG fragments you can create an IntersectionObserver and load pieces only when they get into the viewport. Try create <LazySvg/>

Reply:
You can try const NightLegend = React.lazy(() => import('./NightLegend')); 
and
<Suspense fallback={<p>Loading…</p>}>
  {activeTab === 'night' && <NightLegend />}
</Suspense>